.agents/skills/live-robot-telemetry/scripts/nt_live_common.py

```
@@ -0,0 +1,93 @@
+            return candidate.resolve()
+    raise SystemExit(
+        "Could not discover WPILib root. Set WPILIB_ROOT to the WPILib install directory "
+        "or JAVA_HOME to its packaged jdk."
+    )
+
+
+def wpilib_version(root: Path, group_path: str, artifact: str) -> str:
+    directory = root / "maven" / group_path / artifact
+    versions = sorted((path.name for path in directory.glob("*") if path.is_dir()), reverse=True)
+    if not versions:
+        raise SystemExit(f"Could not find WPILib artifact {artifact} under {directory}")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)